test/integration/apiserver/oidc/oidc_test.go

```diff
@@ -1789,6 +1898,23 @@ func startTestAPIServerForOIDC[L utilsoidc.JosePublicKey](t *testing.T, c apiSer
 	var customFlags []string
 	if len(c.authenticationConfigYAML) > 0 {
 		customFlags = []string{fmt.Sprintf("--authentication-config=%s", writeTempFile(t, c.authenticationConfigYAML))}
+		// super hacky way to also configure egress selector
+		if strings.Contains(c.authenticationConfigYAML, "### NEEDS EGRESS ON START ###") {
```

Comment:
... can we make this a bool on apiServerOIDCConfig instead and toggle that on from the testcases?

Reply:
I was trying to avoid that violence to the test for now, need to refactor the whole thing to make config changes less painful...

Reply:
I thought we were already plumbing a test config struct all the way down